### ISSUE TITLE
Error calling sync triggers (BadRequest)

### ISSUE THREAD

**Issue by chook** [Azure/azure-functions-core-tools#2721]
Error while publishing an eventhub trigger...

version: 3.0.3734

host.json
```
{
  "version": "2.0",
  "logging": {
    "applicationInsights": {
      "samplingSettings": {
        "isEnabled": true,
        "excludedTypes": "Request"
      }
    }
  },
  "extensionBundle": {
    "id": "Microsoft.Azure.Functions.ExtensionBundle",
    "version": "[2.*, 3.0.0)"
  }
}
```

function.json
```
{
  "scriptFile": "../dist/EventHub/index.js",
  "disabled": false,
  "bindings": [
    {
      "name": "eventHub",
      "type": "eventHubTrigger",
      "direction": "in",
      "cardinality": "many",
      "eventHubName": "hub2",
      "connection": "EventHubConnection",
      "consumerGroup": "$Default"
    }
  ]
}

```

```
Request:
Method: POST, RequestUri: 'https://management.azure.com/subscriptions/04154982-c1a6-4a37-af1f-0a6c67615576/resourceGroups/rg1/providers/Microsoft.Web/sites/coralogixblob3/host/default/sync?api-version=2015-08-01', Version: 1.1, Content: System.Net.Http.StringContent, Headers:
{
  Authorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsIng1dCI6Imwzc1EtNTBjQ0g0eEJWWkxIVEd3blNSNzY4MCIsImtpZCI6Imwzc1EtNTBjQ0g0eEJWWkxIVEd3blNSNzY4MCJ9.eyJhdWQiOiJodHRwczovL21hbmFnZW1lbnQuY29yZS53aW5kb3dzLm5ldC8iLCJpc3MiOiJodHRwczovL3N0cy53aW5kb3dzLm5ldC9hMDRjMzg0Ny1lZGM1LTQwMDMtOWVjOS0zZjA4ZTZjYjU5MzAvIiwiaWF0IjoxNjMyMzUzMDUzLCJuYmYiOjE2MzIzNTMwNTMsImV4cCI6MTYzMjM1Njk1MywiYWNyIjoiMSIsImFpbyI6IkFXUUFtLzhUQUFBQXNKZ2doakRqdmlyVHdVSG5uRDM2K3BtTXN5MUlKNW53bDB0Y3FlLzgvMmJUVE5KQ1FVd1BwOGhoWGtqb3ZscjRVOEE1TE5TUFVpQ0tETDIva3lURzgwNE5YbVI5LzdybFF5Vk9pa3VyLzE4cGdjOERLZUVRblNmQXgyU2VYZnI4IiwiYWx0c2VjaWQiOiI1OjoxMDAzMjAwMTZCMjhFODREIiwiYW1yIjpbInB3ZCIsIm1mYSJdLCJhcHBpZCI6IjA0YjA3Nzk1LThkZGItNDYxYS1iYmVlLTAyZjllMWJmN2I0NiIsImFwcGlkYWNyIjoiMCIsImVtYWlsIjoiY2hlbkBjb3JhbG9naXguY29tIiwiZ3JvdXBzIjpbIjE1YzFlM2M0LTI5MzctNDA4OS04MWEzLWFkYTYxZjJmNzhiNiJdLCJpZHAiOiJodHRwczovL3N0cy53aW5kb3dzLm5ldC8xYTY1NTkyZS1jNDg4LTQ4MWEtYWVlOC03NzcxZGUzNzU3YmUvIiwiaXBhZGRyIjoiMTkyLjYzLjE5Ni4xNjEiLCJuYW1lIjoiY2hlbiIsIm9pZCI6IjViZTlkMzIyLTA0NDktNDg0Ny05MzZjLTk4MTdhZTgxMTU5NyIsInB1aWQiOiIxMDAzMjAwMTc4NzlCMDgwIiwicmgiOiIwLkFWQUFSemhNb01YdEEwQ2V5VDhJNXN0Wk1KVjNzQVRialJwR3UtNEMtZUdfZTBaUUFNRS4iLCJzY3AiOiJ1c2VyX2ltcGVyc29uYXRpb24iLCJzdWIiOiJkUmY2VU5BVlFfTXFCOVA4Wm9YaXNqUGg2dGtzb1VBQkR5a0NQQ2VPWkU0IiwidGlkIjoiYTA0YzM4NDctZWRjNS00MDAzLTllYzktM2YwOGU2Y2I1OTMwIiwidW5pcXVlX25hbWUiOiJjaGVuQGNvcmFsb2dpeC5jb20iLCJ1dGkiOiI2eGQxeDZHWTkwR3cxMi1CNGhVT0FBIiwidmVyIjoiMS4wIiwid2lkcyI6WyI2MmU5MDM5NC02OWY1LTQyMzctOTE5MC0wMTIxNzcxNDVlMTAiLCIxM2JkMWM3Mi02ZjRhLTRkY2YtOTg1Zi0xOGQzYjgwZjIwOGEiXSwieG1zX3RjZHQiOjE2Mjg2MzM3MTd9.HexX1RiZZ3NTP3IC0wFkB5z5KyhrwgUuAUtmv5T0USC3FPCx3fledR-sjrdy8_g_xFxnqCEpHNOWP4QHV2NPkEYAJJTLBHU-_zwzmjLMOKAyH1ii0HepfmV2H0rcWsysZpCBGDf3ye5AY5UJ6NV6aqrqTWCDIrUNIte3dmLVDDeV_BypmEBAg8It6Jb9LruBV7j43NoFxEc52JzkesPqwbEKstbMQ7kr1Evgd0DWWgxklQ2J0ErvAoJ_nTa0U1fbTfn42ArnxDmuZ3BQymLdTgRzzou2r47zpny8GUlhy99DQTs-Cj4uep2jIewPNa4YzQD-104XioC9gywm2kKl3A
  User-Agent: functions-core-tools/3.0.3734
  Accept: application/json
  x-ms-request-id: 87df3380-f566-4087-bba1-ee45e02644ad
  Content-Type: application/json; charset=utf-8
}
null

Response:
StatusCode: 400, ReasonPhrase: 'Bad Request', Version: 1.1, Content: System.Net.Http.HttpConnectionResponseContent, Headers:
{
  Cache-Control: no-cache
  Pragma: no-cache
  Strict-Transport-Security: max-age=31536000; includeSubDomains
  Server: Microsoft-IIS/10.0
  X-AspNet-Version: 4.0.30319
  X-Powered-By: ASP.NET
  x-ms-ratelimit-remaining-subscription-writes: 1199
  x-ms-request-id: 405f7def-993b-4903-ba3f-322757783f87
  x-ms-correlation-request-id: 405f7def-993b-4903-ba3f-322757783f87
  x-ms-routing-request-id: SOUTHCENTRALUS:20210922T233437Z:405f7def-993b-4903-ba3f-322757783f87
  X-Content-Type-Options: nosniff
  Date: Wed, 22 Sep 2021 23:34:37 GMT
  Content-Length: 349
  Content-Type: application/json; charset=utf-8
  Expires: -1
}
{"Code":"BadRequest","Message":"Encountered an error (InternalServerError) from host runtime.","Target":null,"Details":[{"Message":"Encountered an error (InternalServerError) from host runtime."},{"Code":"BadRequest"},{"ErrorEntity":{"Code":"BadRequest","Message":"Encountered an error (InternalServerError) from host runtime."}}],"Innererror":null}


Azure.Functions.Cli.Common.CliException: Error calling sync triggers (BadRequest). Request ID = '405f7def-993b-4903-ba3f-322757783f87'.
   at Azure.Functions.Cli.Helpers.RetryHelper.Retry(Func`1 func, Int32 retryCount, TimeSpan retryDelay, Boolean displayError) in D:\a\1\s\src\Azure.Functions.Cli\Helpers\RetryHelper.cs:line 29
   at Azure.Functions.Cli.Actions.AzureActions.PublishFunctionAppAction.SyncTriggers(Site functionApp) in D:\a\1\s\src\Azure.Functions.Cli\Actions\AzureActions\PublishFunctionAppAction.cs:line 493
   at Azure.Functions.Cli.Actions.AzureActions.PublishFunctionAppAction.PublishFunctionApp(Site functionApp, GitIgnoreParser ignoreParser, IDictionary`2 additionalAppSettings) in D:\a\1\s\src\Azure.Functions.Cli\Actions\AzureActions\PublishFunctionAppAction.cs:line 484
   at Azure.Functions.Cli.Actions.AzureActions.PublishFunctionAppAction.RunAsync() in D:\a\1\s\src\Azure.Functions.Cli\Actions\AzureActions\PublishFunctionAppAction.cs:line 176
   at Azure.Functions.Cli.ConsoleApp.RunAsync[T](String[] args, IContainer container) in D:\a\1\s\src\Azure.Functions.Cli\ConsoleApp.cs:line 66
make: *** [publish] Error 1
```

**Comment by apawast** [comment]
@chook can you provide steps on how to reproduce this? Are you able to publish with other types of functions?

**Comment by chook** [comment]
Thanks @apawast ! we have two functions, the blobstorage triggers work as expected.
I have provided here the host.json and function.json which hopefully help.

Is there anywhere in Azure portal a log that may show actionable logs for the bad request?

**Comment by chook** [comment]
super weird. i tried again to validate and now it works .. i guess it was a glitch in Azure.
Thanks for the quick response anyway

**Comment by chook** [comment]
@apawast it seemed that it only started working AFTER we tested the blobstorage trigger, so somehow it "unlocked" the eventhub trigger sync. so we still have an issue here

**Comment by apawast** [comment]
@chook we are not sure if this is reproducible. If this is still an issue for you, it may be best to open a support ticket instead.

**Comment by chook** [comment]
@apawast what is a support ticket? did you try and reproduce this?
I have a colleague who did and reached the same results.

1. The badRequest with sync triggers is clearly a bug on Azure end, as there are multiple questions about it with no clear answers about root cause. Please make an effort to fix your error feedback.
2. Even once we get it to work, by installing the same function name with a Blob trigger, and then move to an Event Hub trigger, it is a false positive. As the installed function never gets triggered.

**Comment by balag0** [comment]
@fabiocav can we add (1) to the scope of https://github.com/Azure/azure-functions-host/issues/4934 ?

**Comment by chook** [comment]
I was able to finally track down the origin, and why installing the blog trigger randomly "fixed" the issue.

**tl;dr my AzureWebJobsStorage config was empty** (the storage trigger set it correctly)


This is going to be a few unrelated things but it's best to document my state of mind and RCA for future generations that may read this:

It started with a working blobstorage trigger I migrated to an eventhub trigger. That worked in the past, but after a few weeks I decided to refactor to split into two separate packages.
Then the eventhub one stopped working. I received the "Error calling sync triggers (BadRequest)" which didn't really say anything.

The first thing I tried to do was to test locally with `npm start` but since I moved to an Apple M1 chip, it took a bit of time to setup node in a correct way that I can actually see what's going on. This is where I noticed this error:
```
Microsoft.Azure.WebJobs.Host: Error indexing method 'Functions.EventHub'. Microsoft.WindowsAzure.Storage: Value cannot be null. (Parameter 'connectionString').
```

Once I realized this ties to the AzureWebJobsStorage configuration I was on the right track!

Bonus:
In order to debug on M1 I had to downgrade to Node 14.16.1 using nvm on an "iTerm Rosseta".

Thanks MSFT for the help!
